appservice/src/SiteClient.ts

```diff
@@ -288,6 +301,58 @@ export class SiteClient {
         const result: string = await requestUtils.sendRequest(requestOptions);
         return <IFunctionKeys>JSON.parse(result);
     }
+
+    /**
+     * To be used for better performance when checking something on the plan that doesn't change
+     */
+    private async getCachedAppServicePlan(): Promise<AppServicePlan | undefined> {
+        let result: AppServicePlan | undefined = this._cachedPlan;
+        if (!result) {
+            result = await this.getAppServicePlan();
+        }
+        return result;
+    }
+
+    /**
+     * Linux consumption currently requires a site-restricted token for all kudu calls (in addition to existing credentials)
+     * This is only a temporary requirement and should be going away eventually
+     */
+    private async getKuduClientOptions(): Promise<ServiceClientOptions | undefined> {
+        if (this.isFunctionApp && this.isLinux) {
+            const isConsumption: boolean = await this.getIsConsumption();
+            if (isConsumption) {
+                // Doing the best we can for types here - 'ms-rest' is severely lacking when it comes to filters
+                type callbackType = (err: unknown) => void;
+                type nextType = (r: WebResource, c: callbackType) => Promise<void>;
+                return {
+                    filters: [
+                        async (resource: WebResource, next: nextType, callback: callbackType): Promise<void> => {
+                            try {
+                                resource.headers['x-ms-site-restricted-token'] = await this.getSiteRestrictedToken();
+                                return next(resource, callback);
+                            } catch (error) {
+                                callback(error);
+                            }
+                        }
+                    ]
+                };
+            }
+        }
+
+        return undefined;
+    }
+
+    private async getSiteRestrictedToken(): Promise<string> {
+        if (!this._cachedSiteRestrictedToken || Date.now() > this._siteRestrictedTokenExpireTime) {
+            const urlPath: string = `${this.id}/hostruntime/admin/host/token?api-version=2015-08-01`;
+            const request: requestUtils.Request = await requestUtils.getDefaultAzureRequest(urlPath, this._subscription);
+            // token only lasts 5 minutes. Use 4 just to be safe
+            this._siteRestrictedTokenExpireTime = Date.now() + 4 * 60 * 1000;
```

Comment:
Azure tokens usually have a number property that specifies when it'll expire.  I think it's called `expiresIn`.  Could just add the exact number that's there rather than making an estimate.

Reply:
This is definitely _not_ your typical azure token. It's a workaround and is _in addition_ to the typical token used for authenticating. The functions team told me to refresh the token based on the 5 minute interval - I never would've known to do it myself.